<!DOCTYPE html> <html><head>
		<title>NormalDataScript</title>
		<base href="../">
		<meta id="root-path" root-path="../">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes, minimum-scale=1.0, maximum-scale=5.0">
		<meta charset="UTF-8">
		<meta name="description" content="LabNotes - NormalDataScript">
		<meta property="og:title" content="NormalDataScript">
		<meta property="og:description" content="LabNotes - NormalDataScript">
		<meta property="og:type" content="website">
		<meta property="og:url" content="scrnaseq/normaldatascript.html">
		<meta property="og:image" content="https://raw.githubusercontent.com/sandelc/obsidian.io/master/Rplot0111.png">
		<meta property="og:site_name" content="LabNotes">
		<link rel="alternate" type="application/rss+xml" title="RSS Feed" href="lib/rss.xml"><script async="" id="webpage-script" src="lib/scripts/webpage.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script type="module" async="" id="graph-view-script" src="lib/scripts/graph-view.js"></script><script async="" id="graph-wasm-script" src="lib/scripts/graph-wasm.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="graph-render-worker-script" src="lib/scripts/graph-render-worker.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="tinycolor-script" src="lib/scripts/tinycolor.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="pixi-script" src="https://cdnjs.cloudflare.com/ajax/libs/pixi.js/7.4.0/pixi.min.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="minisearch-script" src="https://cdn.jsdelivr.net/npm/minisearch@6.3.0/dist/umd/index.min.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><link rel="icon" href="lib/media/favicon.png"><script async="" id="graph-data-script" src="lib/scripts/graph-data.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><style>body{--line-width:40em;--line-width-adaptive:40em;--file-line-width:40em;--sidebar-width:min(20em, 80vw);--collapse-arrow-size:11px;--tree-horizontal-spacing:0.6em;--tree-vertical-spacing:0.6em;--sidebar-margin:12px}.sidebar{height:100%;min-width:calc(var(--sidebar-width) + var(--divider-width-hover));max-width:calc(var(--sidebar-width) + var(--divider-width-hover));font-size:14px;z-index:10;position:relative;overflow:hidden;transition:min-width ease-in-out,max-width ease-in-out;transition-duration:.2s;contain:size}.sidebar-left{left:0}.sidebar-right{right:0}.sidebar.is-collapsed{min-width:0;max-width:0}body.floating-sidebars .sidebar{position:absolute}.sidebar-content{height:100%;min-width:calc(var(--sidebar-width) - var(--divider-width-hover));top:0;padding:var(--sidebar-margin);padding-top:4em;line-height:var(--line-height-tight);background-color:var(--background-secondary);transition:background-color,border-right,border-left,box-shadow;transition-duration:var(--color-fade-speed);transition-timing-function:ease-in-out;position:absolute;display:flex;flex-direction:column}.sidebar:not(.is-collapsed) .sidebar-content{min-width:calc(max(100%,var(--sidebar-width)) - 3px);max-width:calc(max(100%,var(--sidebar-width)) - 3px)}.sidebar-left .sidebar-content{left:0;border-top-right-radius:var(--radius-l);border-bottom-right-radius:var(--radius-l)}.sidebar-right .sidebar-content{right:0;border-top-left-radius:var(--radius-l);border-bottom-left-radius:var(--radius-l)}.sidebar:has(.sidebar-content:empty):has(.topbar-content:empty){display:none}.sidebar-topbar{height:2em;width:var(--sidebar-width);top:var(--sidebar-margin);padding-inline:var(--sidebar-margin);z-index:1;position:fixed;display:flex;align-items:center;transition:width ease-in-out;transition-duration:inherit}.sidebar.is-collapsed .sidebar-topbar{width:calc(2.3em + var(--sidebar-margin) * 2)}.sidebar .sidebar-topbar.is-collapsed{width:0}.sidebar-left .sidebar-topbar{left:0}.sidebar-right .sidebar-topbar{right:0}.topbar-content{overflow:hidden;overflow:clip;width:100%;height:100%;display:flex;align-items:center;transition:inherit}.sidebar.is-collapsed .topbar-content{width:0;transition:inherit}.clickable-icon.sidebar-collapse-icon{background-color:transparent;color:var(--icon-color-focused);padding:0!important;margin:0!important;height:100%!important;width:2.3em!important;margin-inline:0.14em!important;position:absolute}.sidebar-left .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);right:var(--sidebar-margin)}.sidebar-right .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);left:var(--sidebar-margin)}.clickable-icon.sidebar-collapse-icon svg.svg-icon{width:100%;height:100%}.sidebar-section-header{margin:0 0 1em 0;text-transform:uppercase;letter-spacing:.06em;font-weight:600}body{transition:background-color var(--color-fade-speed) ease-in-out}.webpage-container{display:flex;flex-direction:row;height:100%;width:100%;align-items:stretch;justify-content:center}.document-container{opacity:1;flex-basis:100%;max-width:100%;width:100%;height:100%;display:flex;flex-direction:column;align-items:center;transition:opacity .2s ease-in-out;contain:inline-size}.hide{opacity:0;transition:opacity .2s ease-in-out}.document-container>.markdown-preview-view{margin:var(--sidebar-margin);margin-bottom:0;width:100%;width:-webkit-fill-available;width:-moz-available;width:fill-available;background-color:var(--background-primary);transition:background-color var(--color-fade-speed) ease-in-out;border-top-right-radius:var(--window-radius,var(--radius-m));border-top-left-radius:var(--window-radius,var(--radius-m));overflow-x:hidden!important;overflow-y:auto!important;display:flex!important;flex-direction:column!important;align-items:center!important;contain:inline-size}.document-container>.markdown-preview-view>.markdown-preview-sizer{padding-bottom:80vh!important;width:100%!important;max-width:var(--line-width)!important;flex-basis:var(--line-width)!important;transition:background-color var(--color-fade-speed) ease-in-out;contain:inline-size}.markdown-rendered img:not([width]),.view-content img:not([width]){max-width:100%;outline:0}.document-container>.view-content.embed{display:flex;padding:1em;height:100%;width:100%;align-items:center;justify-content:center}.document-container>.view-content.embed>*{max-width:100%;max-height:100%;object-fit:contain}:has(> :is(.math,table)){overflow-x:auto!important}.document-container>.view-content{overflow-x:auto;contain:content;padding:0;margin:0;height:100%}.scroll-highlight{position:absolute;width:100%;height:100%;pointer-events:none;z-index:1000;background-color:hsla(var(--color-accent-hsl),.25);opacity:0;padding:1em;inset:50%;translate:-50% -50%;border-radius:var(--radius-s)}</style><script defer="">async function loadIncludes(){if("file:"!=location.protocol){let e=document.querySelectorAll("include");for(let t=0;t<e.length;t++){let o=e[t],l=o.getAttribute("src");try{const e=await fetch(l);if(!e.ok){console.log("Could not include file: "+l),o?.remove();continue}let t=await e.text(),n=document.createRange().createContextualFragment(t),i=Array.from(n.children);for(let e of i)e.classList.add("hide"),e.style.transition="opacity 0.5s ease-in-out",setTimeout((()=>{e.classList.remove("hide")}),10);o.before(n),o.remove(),console.log("Included file: "+l)}catch(e){o?.remove(),console.log("Could not include file: "+l,e);continue}}}else{if(document.querySelectorAll("include").length>0){var e=document.createElement("div");e.id="error",e.textContent="Web server exports must be hosted on an http / web server to be viewed correctly.",e.style.position="fixed",e.style.top="50%",e.style.left="50%",e.style.transform="translate(-50%, -50%)",e.style.fontSize="1.5em",e.style.fontWeight="bold",e.style.textAlign="center",document.body.appendChild(e),document.querySelector(".document-container")?.classList.remove("hide")}}}document.addEventListener("DOMContentLoaded",(()=>{loadIncludes()}));let isFileProtocol="file:"==location.protocol;function waitLoadScripts(e,t){let o=e.map((e=>document.getElementById(e+"-script"))),l=0;!function e(){let n=o[l];l++,n&&"true"!=n.getAttribute("loaded")||l<o.length&&e(),l<o.length?n.addEventListener("load",e):t()}()}</script><link rel="stylesheet" href="lib/styles/obsidian.css"><link rel="stylesheet" href="lib/styles/theme.css"><link rel="preload" href="lib/styles/global-variable-styles.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/global-variable-styles.css"></noscript><link rel="preload" href="lib/styles/supported-plugins.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/supported-plugins.css"></noscript><link rel="preload" href="lib/styles/main-styles.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/main-styles.css"></noscript></head><body class="publish css-settings-manager theme-dark show-inline-title show-ribbon"><script defer="">let theme=localStorage.getItem("theme")||(window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light");"dark"==theme?(document.body.classList.add("theme-dark"),document.body.classList.remove("theme-light")):(document.body.classList.add("theme-light"),document.body.classList.remove("theme-dark")),window.innerWidth<480?document.body.classList.add("is-phone"):window.innerWidth<768?document.body.classList.add("is-tablet"):window.innerWidth<1024?document.body.classList.add("is-small-screen"):document.body.classList.add("is-large-screen")</script><div class="webpage-container workspace"><div class="sidebar-left sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"><label class="theme-toggle-container" for="theme_toggle"><input class="theme-toggle-input" type="checkbox" id="theme_toggle"><div class="toggle-background"></div></label></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="search-input-container"><input enterkeyhint="search" type="search" spellcheck="false" placeholder="Search..."><div class="search-input-clear-button" aria-label="Clear search"></div></div><include src="lib/html/file-tree.html"></include></div><script defer="">let ls = document.querySelector(".sidebar-left"); ls.classList.add("is-collapsed"); if (window.innerWidth > 768) ls.classList.remove("is-collapsed"); ls.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-left-width"));</script></div><div class="document-container markdown-reading-view hide"><div class="markdown-preview-view markdown-rendered allow-fold-headings allow-fold-lists is-readable-line-width"><style id="MJX-CHTML-styles"></style><div class="markdown-preview-sizer markdown-preview-section"><h1 class="page-title heading inline-title" id="NormalDataScript">NormalDataScript</h1><div class="heading-wrapper"><h3 data-heading="dependencies etc" dir="auto" class="heading" id="dependencies_etc"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>dependencies etc</h3><div class="heading-children"><div><p dir="auto">Seurat Dependencies</p></div><div><pre><code>SeuratPackages &lt;- c('Seurat', 'dplyr','patchwork','glmGamPoi','presto','BPCells','sctransform',               'hdf5r','Azimuth','ggplot2','Signac','SeuratWrappers','BSgenome.Hsapiens.UCSC.hg38')
lapply(SeuratPackages, library, character.only=TRUE)

</code><button class="copy-code-button">Copy</button></pre></div></div></div><div class="heading-wrapper"><h3 data-heading="QC steps" dir="auto" class="heading" id="QC_steps"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>QC steps</h3><div class="heading-children"><div><p dir="auto">Data is processed by cellRanger and imported via the Read10x function</p></div><div><pre class="language-none" tabindex="0"><code class="language-pbmc.data is-loaded language-none">pbmc &lt;- CreateSeuratObject(counts = pbmc.data, project = "control", min.cells = 3, min.features = 200)

pbmc &lt;- CreateSeuratObject(counts = pbmc.data, project = "control", min.cells = 3, min.features = 200)

pbmc[["percent.mt"]] &lt;- PercentageFeatureSet(pbmc, pattern = "^mt-")

VlnPlot(pbmc, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
</code><button class="copy-code-button">Copy</button></pre></div></div></div><div class="heading-wrapper"><h3 data-heading="Steps for heuristics, clustering, marker identification" dir="auto" class="heading" id="Steps_for_heuristics,_clustering,_marker_identification"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Steps for heuristics, clustering, marker identification</h3><div class="heading-children"><div class="heading-wrapper"><h4 data-heading="VlnPlot Output: Features, Counts, PctMt" dir="auto" class="heading" id="VlnPlot_Output:_Features,_Counts,_PctMt"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>VlnPlot Output: Features, Counts, PctMt</h4><div class="heading-children"><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/Rplot0111.png" referrerpolicy="no-referrer"></p></div></div></div><div class="heading-wrapper"><h4 data-heading="FeatureScatter Output: Counts vs  PctMt" dir="auto" class="heading" id="FeatureScatter_Output:_Counts_vs__PctMt"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>FeatureScatter Output: Counts vs  PctMt</h4><div class="heading-children"><div><pre><code>plot1 &lt;- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 &lt;- FeatureScatter(pbmc, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

plot1 + plot2
</code><button class="copy-code-button">Copy</button></pre></div><div class="heading-wrapper"><h5 data-heading="both plots" dir="auto" class="heading" id="both_plots"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>both plots</h5><div class="heading-children"><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/FeatureScatter.png" referrerpolicy="no-referrer"></p></div></div></div><div class="heading-wrapper"><h5 data-heading="plots separate" dir="auto" class="heading" id="plots_separate"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>plots separate</h5><div class="heading-children"><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/FeaturePlotCountsvsPctMt.png" referrerpolicy="no-referrer"></p></div><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/FeaturePlotCountsvsFeatures.png" referrerpolicy="no-referrer"></p></div></div></div></div></div><div class="heading-wrapper"><h4 data-heading="Subset and Normalize" dir="auto" class="heading" id="Subset_and_Normalize"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Subset and Normalize</h4><div class="heading-children"><div><p dir="auto">Examining VlnPlot - ~5% mt cutoff seems appropriate<br>
Looking at featureplots - about 4200 features seems appropriate. could just do an even 4000. </p></div><div><p dir="auto">subsetting: </p></div><div><pre><code>pbmcSubset &lt;- subset(pbmc, subset = nFeature_RNA &gt; 200 &amp; nFeature_RNA &lt; 4000 &amp; percent.mt &lt; 5)
</code><button class="copy-code-button">Copy</button></pre></div><div class="heading-wrapper"><h5 data-heading="VlnPlot after subsetting:" dir="auto" class="heading" id="VlnPlot_after_subsetting:"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>VlnPlot after subsetting:</h5><div class="heading-children"><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/VlnPlotAfterSubset.png" referrerpolicy="no-referrer"></p></div><div><p dir="auto">Normalizing, showing top variable genes:</p></div><div><pre><code>pbmcSubset &lt;- NormalizeData(pbmcSubset, normalization.method = "LogNormalize", scale.factor = 10000)

pbmcSubset &lt;- FindVariableFeatures(pbmcSubset, selection.method = "vst", nfeatures = 2000)
top40 &lt;- head(VariableFeatures(pbmcSubset), 40)
plot1 &lt;- VariableFeaturePlot(pbmcSubset)
plot2 &lt;- LabelPoints(plot = plot1, points = top40, repel = TRUE)
plot2
</code><button class="copy-code-button">Copy</button></pre></div></div></div><div class="heading-wrapper"><h5 data-heading="Top variable genes" dir="auto" class="heading" id="Top_variable_genes"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Top variable genes</h5><div class="heading-children"><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/Top23VariableGenes.png" referrerpolicy="no-referrer"></p></div></div></div></div></div><div class="heading-wrapper"><h4 data-heading="Scaling and Linear Dim Reduction" dir="auto" class="heading" id="Scaling_and_Linear_Dim_Reduction"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Scaling and Linear Dim Reduction</h4><div class="heading-children"><div><pre><code>all.genes &lt;- rownames(pbmcSubset)
pbmcSubset &lt;- ScaleData(pbmcSubset, features = all.genes)
#alternatively, use this to regress out unwanted factors:
pbmcSubset &lt;- ScaleData(pbmcSubset, vars.to.regress = "percent.mt")

pbmcSubset &lt;- RunPCA(pbmcSubset, features = VariableFeatures(object = pbmcSubset))
print(pbmcSubset[["pca"]], dims = 1:5, nfeatures = 5)
</code><button class="copy-code-button">Copy</button></pre></div><div><pre><code>PC_ 1 
Positive:  Gimap3, Sept1, Rpl12, Gimap4, Ms4a4b 
Negative:  Ifitm3, Lyz2, Alox5ap, Lst1, Ifitm2 
PC_ 2 
Positive:  Cd79a, Ebf1, Cd79b, Cd74, Igkc 
Negative:  Nkg7, AW112010, Ms4a4b, Id2, Ccl5 
PC_ 3 
Positive:  Cd300e, Pglyrp1, Adgre4, Fabp4, Ear2 
Negative:  Tmem176b, C1qc, C1qb, C1qa, Cxcl16 
PC_ 4 
Positive:  Cd3d, Cd3e, Il7r, Cd3g, Tcf7 
Negative:  Klrb1c, Ncr1, Gzma, Klre1, Prf1 
PC_ 5 
Positive:  Tmsb10, Fn1, Vim, Crip1, Ccr2 
Negative:  C1qa, C1qc, C1qb, Slco2b1, Cd63
</code><button class="copy-code-button">Copy</button></pre></div><div><pre><code>saveRDS(pbmcSubset, file = "")
</code><button class="copy-code-button">Copy</button></pre></div><div><p dir="auto">Second seurat object with regressed out mt genes</p></div><div><pre><code>pbmcSubsetRegressedMt &lt;- ScaleData(pbmcSubset, vars.to.regress = "percent.mt")
pbmcSubsetRegressedMt &lt;- RunPCA(pbmcSubsetRegressedMt, features = VariableFeatures(object = pbmcSubsetRegressedMt))
</code><button class="copy-code-button">Copy</button></pre></div><div><pre><code>
DimPlot(pbmcSubset, reduction = "pca")

DimPlot(pbmcSubsetRegressedMt, reduction = "pca")
</code><button class="copy-code-button">Copy</button></pre></div><div class="heading-wrapper"><h5 data-heading="PCA" dir="auto" class="heading" id="PCA"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>PCA</h5><div class="heading-children"><div><p dir="auto">PCA on just subsetted data, not regressed for mt<br>
<img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/PCAPbmcSubsetNomtRegression.png" referrerpolicy="no-referrer"></p></div><div><p dir="auto">PCA on regressed data:</p></div><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/PCAregressedMt.png" referrerpolicy="no-referrer"></p></div></div></div><div class="heading-wrapper"><h5 data-heading="elbow plot" dir="auto" class="heading" id="elbow_plot"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>elbow plot</h5><div class="heading-children"><div><p dir="auto">These are used to determine the number of principal components with which to do dimensional reduction</p></div><div><p dir="auto">both analyses (regressed out Mt or no regression) look essentially identical:<br>
this is the regressed dataset<br>
<img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/ElbowPlotRegressedMt.png" referrerpolicy="no-referrer"></p></div><div><p dir="auto">From this it looks like all of the variability is captured up to PC18, although 15 might be a lower cutoff.  proceeding with 18</p></div></div></div></div></div><div class="heading-wrapper"><h4 data-heading="Clustering" dir="auto" class="heading" id="Clustering"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Clustering</h4><div class="heading-children"><div><pre><code>#these perform dimensional reduction with the PC we selected.
#Dimensions are the PC
pbmcSubset &lt;- FindNeighbors(pbmcSubset, dims = 1:18)
pbmcSubset &lt;- FindClusters(pbmcSubset, resolution = 0.5)

pbmcSubsetRegressedMt &lt;- FindNeighbors(pbmcSubsetRegressedMt, dims = 1:18)
pbmcSubsetRegressedMt &lt;- FindClusters(pbmcSubsetRegressedMt, resolution = 0.5)

pbmcSubset &lt;- RunUMAP(pbmcSubset, dims = 1:18)
DimPlot(pbmcSubset, reduction = "umap")
</code><button class="copy-code-button">Copy</button></pre></div><div><p dir="auto">Cluster output for subsetted data no regressed MT:</p></div><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/ClusteringSubsettedButNoRegression.png" referrerpolicy="no-referrer"></p></div><div><pre><code>pbmcSubsetRegressedMt &lt;- RunUMAP(pbmcSubsetRegressedMt, dims = 1:18)
DimPlot(pbmcSubsetRegressedMt, reduction = "umap")
</code><button class="copy-code-button">Copy</button></pre></div><div><p dir="auto">Cluster output for data with regressed mt:<br>
<img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/ClusteringWithRegressedMT.png" referrerpolicy="no-referrer"></p></div></div></div><div class="heading-wrapper"><h4 data-heading="Identify Cluster Biomarkers" dir="auto" class="heading" id="Identify_Cluster_Biomarkers"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Identify Cluster Biomarkers</h4><div class="heading-children"><div><pre><code>#examples of how to find markers in a cluster 
# find all markers of cluster 2
cluster2.markers &lt;- FindMarkers(pbmc, ident.1 = 2)
head(cluster2.markers, n = 5)


# find all markers distinguishing cluster 5 from clusters 0 and 3
cluster0.markers &lt;- FindMarkers(pbmcSubset, ident.1 = 5, ident.2 = c(0, 3))
head(cluster5.markers, n = 5)

# find markers for every cluster compared to all remaining #cells, report only the positive ones
pbmc.markers &lt;- FindAllMarkers(pbmc, only.pos = TRUE)
pbmc.markers %&gt;%
    group_by(cluster) %&gt;%
    dplyr::filter(avg_log2FC &gt; 1)


</code><button class="copy-code-button">Copy</button></pre></div><div><pre><code>cluster0.markers &lt;- FindMarkers(pbmc, ident.1 = 0, logfc.threshold = 0.25, test.use = "roc", only.pos = TRUE)
</code><button class="copy-code-button">Copy</button></pre></div><div><pre><code>#visualize canonical markers across the umap plot
FeaturePlot(pbmc, features = c("MS4A1", "GNLY", "CD3E", "CD14", "FCER1A", "FCGR3A", "LYZ", "PPBP",
    "CD8A"))

</code><button class="copy-code-button">Copy</button></pre></div><div><p dir="auto">working with regressed image:</p></div><div><p dir="auto"><img alt="gh" src="https://raw.githubusercontent.com/sandelc/obsidian.io/master/ClustersRegressedPlotlabeled.png" referrerpolicy="no-referrer"></p></div><div><p dir="auto">cluster#<br>
0<br>
1  Cd79a, Ms4a1 Iglc3 Iglc2... - B cells<br>
2 Cd8a, Cd8b1, Cd3e, Cd3g, - CD8 T cells<br>
3<br>
4 Cd79a -  another B cell cluster<br>
5:  Eomes, Klrg1   - NK cells<br>
6: Cd4, Cd3e, Cd3g - CD4 T cells<br>
7: Cd8a, Cd8b1         - CD8 T cells<br>
8<br>
9<br>
10<br>
11<br>
12</p></div></div></div><div class="heading-wrapper"><h4 data-heading="Using scTransform" dir="auto" class="heading" id="Using_scTransform"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Using scTransform</h4><div class="heading-children"><div><pre><code>#starting with the 10x data imported from cellranger
scPBMC[["percent.mt"]] &lt;- PercentageFeatureSet(pbmc, pattern = "^mt-")
scPBMC &lt;- SCTransform(pbmc, vars.to.regress = "percent.mt", verbose = FALSE)

</code><button class="copy-code-button">Copy</button></pre></div><div><pre><code>pbmc &lt;- RunPCA(pbmc, verbose = FALSE)
pbmc &lt;- RunUMAP(pbmc, dims = 1:30, verbose = FALSE)

pbmc &lt;- FindNeighbors(pbmc, dims = 1:30, verbose = FALSE)
pbmc &lt;- FindClusters(pbmc, verbose = FALSE)
DimPlot(pbmc, label = TRUE)
</code><button class="copy-code-button">Copy</button></pre></div><div><p dir="auto">asdf</p></div><div class="mod-footer"></div></div></div></div></div></div></div></div><div class="sidebar-right sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="graph-view-wrapper"><div class="sidebar-section-header">Interactive Graph</div><div class="graph-view-placeholder">
		<div class="graph-view-container">
			<div class="graph-icon graph-expand" role="button" aria-label="Expand" data-tooltip-position="top"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><line x1="7" y1="17" x2="17" y2="7"></line><polyline points="7 7 17 7 17 17"></polyline></svg></div>
			<canvas id="graph-canvas" class="hide" width="512px" height="512px"></canvas>
		</div>
		</div></div><div class="tree-container mod-root nav-folder tree-item outline-tree" data-depth="0"><div class="tree-header"><span class="sidebar-section-header">Table Of Contents</span><button class="clickable-icon collapse-tree-button" aria-label="Collapse All"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></svg></button></div><div class="tree-scroll-area tree-item-children nav-folder-children"><div class="tree-item mod-tree-folder nav-folder mod-collapsible is-collapsed" style="display: none;"></div><div class="tree-item" data-depth="1"><a class="tree-link" href="scrnaseq/normaldatascript.html#NormalDataScript"><div class="tree-item-contents heading-link" heading-name="NormalDataScript"><span class="tree-item-title">NormalDataScript</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="3"><a class="tree-link" href="scrnaseq/normaldatascript.html#dependencies_etc"><div class="tree-item-contents heading-link" heading-name="dependencies etc"><span class="tree-item-title">dependencies etc</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="scrnaseq/normaldatascript.html#QC_steps"><div class="tree-item-contents heading-link" heading-name="QC steps"><span class="tree-item-title">QC steps</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="scrnaseq/normaldatascript.html#Steps_for_heuristics,_clustering,_marker_identification"><div class="tree-item-contents heading-link" heading-name="Steps for heuristics, clustering, marker identification"><span class="tree-item-title">Steps for heuristics, clustering, marker identification</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#VlnPlot_Output:_Features,_Counts,_PctMt"><div class="tree-item-contents heading-link" heading-name="VlnPlot Output: Features, Counts, PctMt"><span class="tree-item-title">VlnPlot Output: Features, Counts, PctMt</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#FeatureScatter_Output:_Counts_vs__PctMt"><div class="tree-item-contents heading-link" heading-name="FeatureScatter Output: Counts vs  PctMt"><span class="tree-item-title">FeatureScatter Output: Counts vs  PctMt</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="scrnaseq/normaldatascript.html#both_plots"><div class="tree-item-contents heading-link" heading-name="both plots"><span class="tree-item-title">both plots</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="scrnaseq/normaldatascript.html#plots_separate"><div class="tree-item-contents heading-link" heading-name="plots separate"><span class="tree-item-title">plots separate</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#Subset_and_Normalize"><div class="tree-item-contents heading-link" heading-name="Subset and Normalize"><span class="tree-item-title">Subset and Normalize</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="scrnaseq/normaldatascript.html#VlnPlot_after_subsetting:"><div class="tree-item-contents heading-link" heading-name="VlnPlot after subsetting:"><span class="tree-item-title">VlnPlot after subsetting:</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="scrnaseq/normaldatascript.html#Top_variable_genes"><div class="tree-item-contents heading-link" heading-name="Top variable genes"><span class="tree-item-title">Top variable genes</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#Scaling_and_Linear_Dim_Reduction"><div class="tree-item-contents heading-link" heading-name="Scaling and Linear Dim Reduction"><span class="tree-item-title">Scaling and Linear Dim Reduction</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="scrnaseq/normaldatascript.html#PCA"><div class="tree-item-contents heading-link" heading-name="PCA"><span class="tree-item-title">PCA</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="5"><a class="tree-link" href="scrnaseq/normaldatascript.html#elbow_plot"><div class="tree-item-contents heading-link" heading-name="elbow plot"><span class="tree-item-title">elbow plot</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#Clustering"><div class="tree-item-contents heading-link" heading-name="Clustering"><span class="tree-item-title">Clustering</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#Identify_Cluster_Biomarkers"><div class="tree-item-contents heading-link" heading-name="Identify Cluster Biomarkers"><span class="tree-item-title">Identify Cluster Biomarkers</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="scrnaseq/normaldatascript.html#Using_scTransform"><div class="tree-item-contents heading-link" heading-name="Using scTransform"><span class="tree-item-title">Using scTransform</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div></div><script defer="">let rs = document.querySelector(".sidebar-right"); rs.classList.add("is-collapsed"); if (window.innerWidth > 768) rs.classList.remove("is-collapsed"); rs.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-right-width"));</script></div></div></body></html>