<!DOCTYPE html> <html><head>
		<title>BestPractices</title>
		<base href="../../../">
		<meta id="root-path" root-path="../../../">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes, minimum-scale=1.0, maximum-scale=5.0">
		<meta charset="UTF-8">
		<meta name="description" content="LabNotes - BestPractices">
		<meta property="og:title" content="BestPractices">
		<meta property="og:description" content="LabNotes - BestPractices">
		<meta property="og:type" content="website">
		<meta property="og:url" content="notes/scrnaseqnotes/generalconcepts/bestpractices.html">
		<meta property="og:image" content="undefined">
		<meta property="og:site_name" content="LabNotes">
		<link rel="alternate" type="application/rss+xml" title="RSS Feed" href="lib/rss.xml"><script async="" id="webpage-script" src="lib/scripts/webpage.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script type="module" async="" id="graph-view-script" src="lib/scripts/graph-view.js"></script><script async="" id="graph-wasm-script" src="lib/scripts/graph-wasm.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="graph-render-worker-script" src="lib/scripts/graph-render-worker.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="tinycolor-script" src="lib/scripts/tinycolor.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="pixi-script" src="https://cdnjs.cloudflare.com/ajax/libs/pixi.js/7.4.0/pixi.min.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><script async="" id="minisearch-script" src="https://cdn.jsdelivr.net/npm/minisearch@6.3.0/dist/umd/index.min.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><link rel="icon" href="lib/media/favicon.png"><script async="" id="graph-data-script" src="lib/scripts/graph-data.js" onload="this.onload=null;this.setAttribute(&quot;loaded&quot;, &quot;true&quot;)"></script><style>body{--line-width:40em;--line-width-adaptive:40em;--file-line-width:40em;--sidebar-width:min(20em, 80vw);--collapse-arrow-size:11px;--tree-horizontal-spacing:0.6em;--tree-vertical-spacing:0.6em;--sidebar-margin:12px}.sidebar{height:100%;min-width:calc(var(--sidebar-width) + var(--divider-width-hover));max-width:calc(var(--sidebar-width) + var(--divider-width-hover));font-size:14px;z-index:10;position:relative;overflow:hidden;transition:min-width ease-in-out,max-width ease-in-out;transition-duration:.2s;contain:size}.sidebar-left{left:0}.sidebar-right{right:0}.sidebar.is-collapsed{min-width:0;max-width:0}body.floating-sidebars .sidebar{position:absolute}.sidebar-content{height:100%;min-width:calc(var(--sidebar-width) - var(--divider-width-hover));top:0;padding:var(--sidebar-margin);padding-top:4em;line-height:var(--line-height-tight);background-color:var(--background-secondary);transition:background-color,border-right,border-left,box-shadow;transition-duration:var(--color-fade-speed);transition-timing-function:ease-in-out;position:absolute;display:flex;flex-direction:column}.sidebar:not(.is-collapsed) .sidebar-content{min-width:calc(max(100%,var(--sidebar-width)) - 3px);max-width:calc(max(100%,var(--sidebar-width)) - 3px)}.sidebar-left .sidebar-content{left:0;border-top-right-radius:var(--radius-l);border-bottom-right-radius:var(--radius-l)}.sidebar-right .sidebar-content{right:0;border-top-left-radius:var(--radius-l);border-bottom-left-radius:var(--radius-l)}.sidebar:has(.sidebar-content:empty):has(.topbar-content:empty){display:none}.sidebar-topbar{height:2em;width:var(--sidebar-width);top:var(--sidebar-margin);padding-inline:var(--sidebar-margin);z-index:1;position:fixed;display:flex;align-items:center;transition:width ease-in-out;transition-duration:inherit}.sidebar.is-collapsed .sidebar-topbar{width:calc(2.3em + var(--sidebar-margin) * 2)}.sidebar .sidebar-topbar.is-collapsed{width:0}.sidebar-left .sidebar-topbar{left:0}.sidebar-right .sidebar-topbar{right:0}.topbar-content{overflow:hidden;overflow:clip;width:100%;height:100%;display:flex;align-items:center;transition:inherit}.sidebar.is-collapsed .topbar-content{width:0;transition:inherit}.clickable-icon.sidebar-collapse-icon{background-color:transparent;color:var(--icon-color-focused);padding:0!important;margin:0!important;height:100%!important;width:2.3em!important;margin-inline:0.14em!important;position:absolute}.sidebar-left .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);right:var(--sidebar-margin)}.sidebar-right .clickable-icon.sidebar-collapse-icon{transform:rotateY(180deg);left:var(--sidebar-margin)}.clickable-icon.sidebar-collapse-icon svg.svg-icon{width:100%;height:100%}.sidebar-section-header{margin:0 0 1em 0;text-transform:uppercase;letter-spacing:.06em;font-weight:600}body{transition:background-color var(--color-fade-speed) ease-in-out}.webpage-container{display:flex;flex-direction:row;height:100%;width:100%;align-items:stretch;justify-content:center}.document-container{opacity:1;flex-basis:100%;max-width:100%;width:100%;height:100%;display:flex;flex-direction:column;align-items:center;transition:opacity .2s ease-in-out;contain:inline-size}.hide{opacity:0;transition:opacity .2s ease-in-out}.document-container>.markdown-preview-view{margin:var(--sidebar-margin);margin-bottom:0;width:100%;width:-webkit-fill-available;width:-moz-available;width:fill-available;background-color:var(--background-primary);transition:background-color var(--color-fade-speed) ease-in-out;border-top-right-radius:var(--window-radius,var(--radius-m));border-top-left-radius:var(--window-radius,var(--radius-m));overflow-x:hidden!important;overflow-y:auto!important;display:flex!important;flex-direction:column!important;align-items:center!important;contain:inline-size}.document-container>.markdown-preview-view>.markdown-preview-sizer{padding-bottom:80vh!important;width:100%!important;max-width:var(--line-width)!important;flex-basis:var(--line-width)!important;transition:background-color var(--color-fade-speed) ease-in-out;contain:inline-size}.markdown-rendered img:not([width]),.view-content img:not([width]){max-width:100%;outline:0}.document-container>.view-content.embed{display:flex;padding:1em;height:100%;width:100%;align-items:center;justify-content:center}.document-container>.view-content.embed>*{max-width:100%;max-height:100%;object-fit:contain}:has(> :is(.math,table)){overflow-x:auto!important}.document-container>.view-content{overflow-x:auto;contain:content;padding:0;margin:0;height:100%}.scroll-highlight{position:absolute;width:100%;height:100%;pointer-events:none;z-index:1000;background-color:hsla(var(--color-accent-hsl),.25);opacity:0;padding:1em;inset:50%;translate:-50% -50%;border-radius:var(--radius-s)}</style><script defer="">async function loadIncludes(){if("file:"!=location.protocol){let e=document.querySelectorAll("include");for(let t=0;t<e.length;t++){let o=e[t],l=o.getAttribute("src");try{const e=await fetch(l);if(!e.ok){console.log("Could not include file: "+l),o?.remove();continue}let t=await e.text(),n=document.createRange().createContextualFragment(t),i=Array.from(n.children);for(let e of i)e.classList.add("hide"),e.style.transition="opacity 0.5s ease-in-out",setTimeout((()=>{e.classList.remove("hide")}),10);o.before(n),o.remove(),console.log("Included file: "+l)}catch(e){o?.remove(),console.log("Could not include file: "+l,e);continue}}}else{if(document.querySelectorAll("include").length>0){var e=document.createElement("div");e.id="error",e.textContent="Web server exports must be hosted on an http / web server to be viewed correctly.",e.style.position="fixed",e.style.top="50%",e.style.left="50%",e.style.transform="translate(-50%, -50%)",e.style.fontSize="1.5em",e.style.fontWeight="bold",e.style.textAlign="center",document.body.appendChild(e),document.querySelector(".document-container")?.classList.remove("hide")}}}document.addEventListener("DOMContentLoaded",(()=>{loadIncludes()}));let isFileProtocol="file:"==location.protocol;function waitLoadScripts(e,t){let o=e.map((e=>document.getElementById(e+"-script"))),l=0;!function e(){let n=o[l];l++,n&&"true"!=n.getAttribute("loaded")||l<o.length&&e(),l<o.length?n.addEventListener("load",e):t()}()}</script><link rel="stylesheet" href="lib/styles/obsidian.css"><link rel="stylesheet" href="lib/styles/theme.css"><link rel="preload" href="lib/styles/global-variable-styles.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/global-variable-styles.css"></noscript><link rel="preload" href="lib/styles/supported-plugins.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/supported-plugins.css"></noscript><link rel="preload" href="lib/styles/main-styles.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><noscript><link rel="stylesheet" href="lib/styles/main-styles.css"></noscript></head><body class="publish css-settings-manager theme-dark show-inline-title show-ribbon"><script defer="">let theme=localStorage.getItem("theme")||(window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light");"dark"==theme?(document.body.classList.add("theme-dark"),document.body.classList.remove("theme-light")):(document.body.classList.add("theme-light"),document.body.classList.remove("theme-dark")),window.innerWidth<480?document.body.classList.add("is-phone"):window.innerWidth<768?document.body.classList.add("is-tablet"):window.innerWidth<1024?document.body.classList.add("is-small-screen"):document.body.classList.add("is-large-screen")</script><div class="webpage-container workspace"><div class="sidebar-left sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"><label class="theme-toggle-container" for="theme_toggle"><input class="theme-toggle-input" type="checkbox" id="theme_toggle"><div class="toggle-background"></div></label></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="search-input-container"><input enterkeyhint="search" type="search" spellcheck="false" placeholder="Search..."><div class="search-input-clear-button" aria-label="Clear search"></div></div><include src="lib/html/file-tree.html"></include></div><script defer="">let ls = document.querySelector(".sidebar-left"); ls.classList.add("is-collapsed"); if (window.innerWidth > 768) ls.classList.remove("is-collapsed"); ls.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-left-width"));</script></div><div class="document-container markdown-reading-view hide"><div class="markdown-preview-view markdown-rendered allow-fold-headings allow-fold-lists is-readable-line-width"><style id="MJX-CHTML-styles"></style><div class="markdown-preview-sizer markdown-preview-section"><h1 class="page-title heading inline-title" id="BestPractices">BestPractices</h1><div><p dir="auto"><a data-tooltip-position="top" aria-label="https://www.embopress.org/doi/full/10.15252/msb.20188746" rel="noopener" class="external-link" href="https://www.embopress.org/doi/full/10.15252/msb.20188746" target="_blank">BestPracticesTutorialPublication</a><br>
notes follow - </p></div><div><p dir="auto">raw data - CellRanger, indrops, zUMIs do this.<br>
Assigning reads to cell barcodes --- "Demultiplexing"<br>
gene alignment, quantifiation<br>
yields read/count matrix of BARCODES vs genes.<br>
BC some barcodes may be tagged on more than one cell (doublet)<br>
or no cells (empty droplet, empty well)</p></div><div><p dir="auto">QC Metrics - (coviariates)</p></div><div><ul>
<li data-line="0" dir="auto">Counts per barcode</li>
<li data-line="1" dir="auto">Genes per barcode</li>
<li data-line="2" dir="auto">%mt genes per barcode<br>
<a data-tooltip-position="top" aria-label="https://pubmed.ncbi.nlm.nih.gov/26887813/" rel="noopener" class="external-link" href="https://pubmed.ncbi.nlm.nih.gov/26887813/" target="_blank">classification of low quality cells from scrnaseq</a></li>
</ul></div><div><p dir="auto">Other QC Metrics</p></div><div><ul>
<li data-line="0" dir="auto">Read Quality - Filter on reads with over 30 in Q30 scores</li>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Datasets with low Q30 scores in barcode read have poor cell assignment
<ul>
<li data-line="2" dir="auto">poor Q30 scores in the RNA read will not be able to be aligned</li>
<li data-line="3" dir="auto">Typical values 60-70% throughout, with highest in the barcode read</li>
</ul>
</li>
<li data-line="4" dir="auto">Alignment QC - 40% or higher exonic reads is good</li>
<li data-line="5" dir="auto">Direct calculation ? - the % of reads that are in the dataset after count matrix is made. Can estimate the cell input (hemacytometer counts, flow counts) If we get around the same number of cells in a count matrix, amazing. </li>
</ul></div><div><p dir="auto">QC metrics must be considered together, not in isolation. EG high mt counts may suggest involvement in respiration.  Cells/barcodes with low counts may be quiescent cells, high counts may be physically large cells.<br>
Then how to make thresholding decisions?<br>
Be permissive.<br>
Consider whether a threshold removes more than the expected # of cells</p></div><div><p dir="auto">Datasets with mixed cell populations may have multiple peaks in QC metrics</p></div><div><ul>
<li data-line="0" dir="auto">Addressing QC parameters in an analysis workflow changes the output of downstream analysis methods ie pathways, clustering etc. </li>
<li data-line="1" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Knowing whether the QC filters/methods applied are adequate is impossible to know, but can be inferred by performance of downstream analysis methods. 
<ul>
<li data-line="2" dir="auto">Therefore, recursive analysis is required.</li>
</ul>
</li>
</ul></div><div><p dir="auto"><a data-tooltip-position="top" aria-label="https://www.nature.com/articles/s41586-018-0394-6" rel="noopener" class="external-link" href="https://www.nature.com/articles/s41586-018-0394-6" target="_blank">Determine QC metrics independently across samples as in this example</a></p></div><div class="heading-wrapper"><h3 data-heading="doublet detection" dir="auto" class="heading" id="doublet_detection"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>doublet detection</h3><div class="heading-children"><div><p dir="auto">the simple approach is de-select cells that have unusually high gene counts.  In practice what does that mean i'm not sure.</p></div><div><p dir="auto"><a data-tooltip-position="top" aria-label="https://www.cell.com/cell-reports/fulltext/S2211-1247(19)31286-0" rel="noopener" class="external-link" href="https://www.cell.com/cell-reports/fulltext/S2211-1247(19)31286-0" target="_blank">DoubletDecon</a> is one doublet detection method<br>
<a data-tooltip-position="top" aria-label="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6625319/" rel="noopener" class="external-link" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6625319/" target="_blank">Scrublet</a><br>
<a data-tooltip-position="top" aria-label="https://pubmed.ncbi.nlm.nih.gov/30954475/" rel="noopener" class="external-link" href="https://pubmed.ncbi.nlm.nih.gov/30954475/" target="_blank">DoubletFinder</a></p></div></div></div><div class="heading-wrapper"><h3 data-heading="Ambient RNA" dir="auto" class="heading" id="Ambient_RNA"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Ambient RNA</h3><div class="heading-children"><div><p dir="auto"><a data-tooltip-position="top" aria-label="https://www.biorxiv.org/content/10.1101/303727v2" rel="noopener" class="external-link" href="https://www.biorxiv.org/content/10.1101/303727v2" target="_blank">soupx</a><br>
also (<a rel="noopener" class="external-link" href="https://academic.oup.com/gigascience/article/9/12/giaa151/6049831" target="_blank">https://academic.oup.com/gigascience/article/9/12/giaa151/6049831</a>)<br>
removes ambient rna by analyzing empty droplets?????</p></div></div></div><div class="heading-wrapper"><h3 data-heading="Normalization" dir="auto" class="heading" id="Normalization"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Normalization</h3><div class="heading-children"><div><p dir="auto">The wetlab process has a lot of room for error.  Cell digestion, isolation, sorting, capture/bead, RT, sequencing.<br>
Count depth can vary greatly simply due to these processes, even when processed the same.  Therefore identical cells could potentially look very different.<br>
Normalization is required to address this</p></div><div class="heading-wrapper"><h4 data-heading="Linear" dir="auto" class="heading" id="Linear"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Linear</h4><div class="heading-children"><div><p dir="auto">a simple way is to scale the gene expression per cell by the total library size<br>
CPM normalization. or scaling by the median count depth per cell<br>
the assumption here is that all cells initially had equal RNA and the only difference in RELATIVE amounts is due to sampling error.<br>
Downsampling has the same assumption. It randomly samples reads/counts so that all cells have a certain # of counts or less. (deletes data)<br>
These approaches aren't great for mixed populations.  Maybe they are good for very uniform populations<br>
<a data-tooltip-position="top" aria-label="https://pubmed.ncbi.nlm.nih.gov/27122128/" rel="noopener" class="external-link" href="https://pubmed.ncbi.nlm.nih.gov/27122128/" target="_blank">this pooling method</a>  (scran) does a linear regression of genes after pooling all the cells, to estimate a size factor. and available <a data-tooltip-position="top" aria-label="https://bioconductor.org/packages/release/bioc/html/scran.html" rel="noopener" class="external-link" href="https://bioconductor.org/packages/release/bioc/html/scran.html" target="_blank">here?</a><br>
Scran outperforms other normalization methods <a data-tooltip-position="top" aria-label="https://doi.org/10.1038/s41592-018-0254-1" rel="noopener" class="external-link" href="https://doi.org/10.1038/s41592-018-0254-1" target="_blank">one example in batch correction</a> and <a data-tooltip-position="top" aria-label="https://www.biorxiv.org/content/10.1101/583013v3" rel="noopener" class="external-link" href="https://www.biorxiv.org/content/10.1101/583013v3" target="_blank">DE analysis</a></p></div><div><p dir="auto">Normalization with bulk methods was used at first but zero-inflated sc Data required new approaches<br>
<a data-tooltip-position="top" aria-label="https://pubmed.ncbi.nlm.nih.gov/27122128/" rel="noopener" class="external-link" href="https://pubmed.ncbi.nlm.nih.gov/27122128/" target="_blank">one example</a> and a <a data-tooltip-position="top" aria-label="https://www.nature.com/articles/nmeth.4292" rel="noopener" class="external-link" href="https://www.nature.com/articles/nmeth.4292" target="_blank">second one</a> <a rel="noopener" class="external-link" href="https://doi.org/10.1038/nmeth.4292" target="_blank">https://doi.org/10.1038/nmeth.4292</a></p></div></div></div><div class="heading-wrapper"><h4 data-heading="non-linear" dir="auto" class="heading" id="non-linear"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>non-linear</h4><div class="heading-children"><div><p dir="auto">Can account for more complex unwanted variation.<br>
examples from <a data-tooltip-position="top" aria-label="https://pubmed.ncbi.nlm.nih.gov/29513653/" rel="noopener" class="external-link" href="https://pubmed.ncbi.nlm.nih.gov/29513653/" target="_blank">Satijas? group</a></p></div></div></div><div class="heading-wrapper"><h4 data-heading="Normalization over Genes vs Cells" dir="auto" class="heading" id="Normalization_over_Genes_vs_Cells"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Normalization over Genes vs Cells</h4><div class="heading-children"><div><p dir="auto">All these normalizations above apply to cells...  Normalization could be applied to genes.<br>
The central question is - should all genes be weighted equally?</p></div><div><ul>
<li data-line="0" dir="auto">put the way the paper i'm following today  - "is the magnitude of gene expression informative for the importance of the gene"<br>
can evaluate normalization tools with <a data-tooltip-position="top" aria-label="https://www.cell.com/cell-systems/fulltext/S2405-4712(19)30080-8" rel="noopener" class="external-link" href="https://www.cell.com/cell-systems/fulltext/S2405-4712(19)30080-8" target="_blank">scone</a></li>
</ul></div></div></div><div class="heading-wrapper"><h4 data-heading="transformation" dir="auto" class="heading" id="transformation"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>transformation</h4><div class="heading-children"><div><ul>
<li data-line="0" dir="auto"><div class="list-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>3 important effects
<ul>
<li data-line="1" dir="auto">Distance between log-txformed expression values become logFC, which is what we are typically trained to evaluate</li>
<li data-line="2" dir="auto">log-txformation mitigates the mean-variance relationship in single cell data <a data-tooltip-position="top" aria-label="https://pubmed.ncbi.nlm.nih.gov/24056876/" rel="noopener" class="external-link" href="https://pubmed.ncbi.nlm.nih.gov/24056876/" target="_blank">by reducing technical noise?</a></li>
<li data-line="3" dir="auto">reduces the skewness of the data, so that assumptions of normality can be (sorta) met for downstream tools</li>
</ul>
</li>
<li data-line="4" dir="auto">SCRNASEQ is not lognormal data <a data-tooltip-position="top" aria-label="https://academic.oup.com/bioinformatics/article/33/21/3486/3952669" rel="noopener" class="external-link" href="https://academic.oup.com/bioinformatics/article/33/21/3486/3952669" target="_blank">powsimrPaperDiscusses?</a></li>
<li data-line="5" dir="auto">Log transformation <a data-tooltip-position="top" aria-label="https://www.biorxiv.org/content/10.1101/404962v1" rel="noopener" class="external-link" href="https://www.biorxiv.org/content/10.1101/404962v1" target="_blank">can introduce error</a></li>
</ul></div></div></div></div></div><div class="heading-wrapper"><h3 data-heading="Annotation" dir="auto" class="heading" id="Annotation"><div class="heading-collapse-indicator collapse-indicator collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon right-triangle"><path d="M3 8L12 17L21 8"></path></svg></div>Annotation</h3><div class="heading-children"><div><p dir="auto"><a data-tooltip-position="top" aria-label="https://www.nature.com/articles/nmeth.4644" rel="noopener" class="external-link" href="https://www.nature.com/articles/nmeth.4644" target="_blank">scmap</a><br>
<a rel="noopener" class="external-link" href="https://doi.org/10.1038/nmeth.4644" target="_blank">https://doi.org/10.1038/nmeth.4644</a></p></div><div class="mod-footer"></div></div></div></div></div></div><div class="sidebar-right sidebar"><div class="sidebar-handle"></div><div class="sidebar-topbar"><div class="topbar-content"></div><div class="clickable-icon sidebar-collapse-icon"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><path d="M21 3H3C1.89543 3 1 3.89543 1 5V19C1 20.1046 1.89543 21 3 21H21C22.1046 21 23 20.1046 23 19V5C23 3.89543 22.1046 3 21 3Z"></path><path d="M10 4V20"></path><path d="M4 7H7"></path><path d="M4 10H7"></path><path d="M4 13H7"></path></svg></div></div><div class="sidebar-content"><div class="graph-view-wrapper"><div class="sidebar-section-header">Interactive Graph</div><div class="graph-view-placeholder">
		<div class="graph-view-container">
			<div class="graph-icon graph-expand" role="button" aria-label="Expand" data-tooltip-position="top"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="svg-icon"><line x1="7" y1="17" x2="17" y2="7"></line><polyline points="7 7 17 7 17 17"></polyline></svg></div>
			<canvas id="graph-canvas" class="hide" width="512px" height="512px"></canvas>
		</div>
		</div></div><div class="tree-container mod-root nav-folder tree-item outline-tree" data-depth="0"><div class="tree-header"><span class="sidebar-section-header">Table Of Contents</span><button class="clickable-icon collapse-tree-button" aria-label="Collapse All"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></svg></button></div><div class="tree-scroll-area tree-item-children nav-folder-children"><div class="tree-item mod-tree-folder nav-folder mod-collapsible is-collapsed" style="display: none;"></div><div class="tree-item" data-depth="1"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#BestPractices"><div class="tree-item-contents heading-link" heading-name="BestPractices"><span class="tree-item-title">BestPractices</span></div></a><div class="tree-item-children nav-folder-children"><div class="tree-item" data-depth="3"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#doublet_detection"><div class="tree-item-contents heading-link" heading-name="doublet detection"><span class="tree-item-title">doublet detection</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#Ambient_RNA"><div class="tree-item-contents heading-link" heading-name="Ambient RNA"><span class="tree-item-title">Ambient RNA</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#Normalization"><div class="tree-item-contents heading-link" heading-name="Normalization"><span class="tree-item-title">Normalization</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#Linear"><div class="tree-item-contents heading-link" heading-name="Linear"><span class="tree-item-title">Linear</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#non-linear"><div class="tree-item-contents heading-link" heading-name="non-linear"><span class="tree-item-title">non-linear</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#Normalization_over_Genes_vs_Cells"><div class="tree-item-contents heading-link" heading-name="Normalization over Genes vs Cells"><span class="tree-item-title">Normalization over Genes vs Cells</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="4"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#transformation"><div class="tree-item-contents heading-link" heading-name="transformation"><span class="tree-item-title">transformation</span></div></a><div class="tree-item-children nav-folder-children"></div></div><div class="tree-item" data-depth="3"><a class="tree-link" href="notes/scrnaseqnotes/generalconcepts/bestpractices.html#Annotation"><div class="tree-item-contents heading-link" heading-name="Annotation"><span class="tree-item-title">Annotation</span></div></a><div class="tree-item-children nav-folder-children"></div></div></div></div></div></div></div><script defer="">let rs = document.querySelector(".sidebar-right"); rs.classList.add("is-collapsed"); if (window.innerWidth > 768) rs.classList.remove("is-collapsed"); rs.style.setProperty("--sidebar-width", localStorage.getItem("sidebar-right-width"));</script></div></div></body></html>